### ISSUE TITLE
ODP-RP04 - Support language change after login

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#1987]
Reference [CATS document](https://canada-ca.github.io/CATS-STAE/oidc1-en.pdf) section 5.2.3
```
RP implementations SHOULD examine the value of the locale claim returned by the OP in
order to determine whether the user changed their preferred official language while interacting
with the OP. The RP application SHOULD then display all subsequent content in the newly
selected language.
```

Acceptance Criteria:
- [ ] If the user is sent to SiC in one language and changes the language there, the app should respect that choice when they return
    - See Matt's _[Test scenario 3](https://github.com/GCTC-NTGC/gc-digital-talent/issues/1987#issuecomment-1533528627)_ for an example

**Comment by mnigh** [comment]
@petertgiles does Sign In Canada even return a locale?

**Comment by petertgiles** [comment]
> @petertgiles does Sign In Canada even return a locale?

I think they do.  And I think we do this already?  Feel free to test...

**Comment by mnigh** [comment]
@petertgiles is **Test Scenario 3** the intended outcome or is should changing the locale on an external page of Sign In Canada/GCKey override what was set on talent.canada.ca?


## Test Scenario 1
1. https://talent.canada.ca/en/
2. Logged in via GCKey
3. Returned by Sign In Canada to talent.canada.ca in English

## Test Scenario 2
1. https://talent.canada.ca/fr/
2. Logged in via GCKey
3. Returned by Sign In Canada to talent.canada.ca in French

## Test Scenario 3
1. https://talent.canada.ca/en/
2. Logged in via GCKey
3. Changed language of GCKey page to French
4. Returned by Sign In Canada to talent.canada.ca in English

**Comment by petertgiles** [comment]
Oh yeah, it's a problem.  We should respect the language that SIC used.  In scenario 3 we should switch to French when they get back.

**Comment by esizer** [comment]
Is this ready to be thawed?

**Comment by petertgiles** [comment]
> Is this ready to be thawed?

Good idea.  I added AC and sent it to the refinery.

**Comment by esizer** [comment]
Is there a link where we could read some documentation about the payload we get in the callback? I tried logging it and the best I could get was reading laravel's locale and not the locale from the origin. 😢 

`$request->getLocale();` is always `en` but that is because I believe it is from the Laravel config.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/3e2bea4511a1d75f10241fd55e7704d4a7d93d1d/api/config/app.php#L95

**Comment by petertgiles** [comment]
> Is there a link where we could read some documentation about the payload we get in the callback?

Unfortunately, not that I know of.  :grimacing: I set up my environment to use the test server and grabbed an ID token.
https://github.com/GCTC-NTGC/gc-digital-talent/blob/ec33233067ace50822ad40bcb2bf59bbb11ab2b0/api/app/Http/Controllers/AuthController.php#L80
When I decoded it on [jwt.io](jwt.io) I could see a locale provided.
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/fb944ab2-f6b4-4a5c-a29d-06d75171a0f2)

Yay!

Then I tried it in French at it returned `  "locale": "en-CA en",`  .
So, it changed but not the way I expected.  The first step might be checking with SiC team to see if they need to do some configuration on their side.

**Comment by petertgiles** [comment]
Sign in Canada team confirmed that changing the language in GCKey will not result in a change to the locale field in the ID token.  I'm not sure we have any way to complete this.

**Comment by esizer** [comment]
> Sign in Canada team confirmed that changing the language in GCKey will not result in a change to the locale field in the ID token. I'm not sure we have any way to complete this.

Shoot! 😢

**Comment by mnigh** [comment]
> Sign in Canada team confirmed that changing the language in GCKey will not result in a change to the locale field in the ID token. I'm not sure we have any way to complete this.

@petertgiles did they indicate whether or not this would be a feature that they would be adding in the future?

**Comment by petertgiles** [comment]
> did they indicate whether or not this would be a feature that they would be adding in the future

No, they didn't mention any future plans.

**Comment by mnigh** [comment]
@tristan-orourke can we close this issue as won't fix since this is out of our control?

**Comment by tristan-orourke** [comment]
Lol, if they don't notify us of the language change then this is dead in the water. Closing!

**Comment by petertgiles** [comment]
Further details from the SiC team:
> The locale claim should normally reflect the user’s most recently selected language, however GCKey and CBS pass the language back to us using a third party cookie which stopped working several years back when browsers started blocking these by default. So if the user switches language on a GCKey, Interac or bank sign in page then the change is not passed back as it should be.
